diagnostics/networking_diagnostics.bat

```diff
@@ -0,0 +1,60 @@
+:: Check for administrator privileges.
+net session >nul 2>&1 || goto :admin
+
+:: Validate that required files are here.
+if not exist wsl_networking.wprp (echo wsl_networking.wprp not found && exit /b 1)
+if not exist networking.sh (echo networking.sh not found && exit /b 1)
+
+set networking_folder=".\networking_logs"
+set neworking_logs_zip=".\WslNetworkingLogs.zip"
+
+mkdir %networking_folder%
+
+cd %networking_folder%
+
+IF "%1"=="--stop-wsl" (
+   :: The WSL HNS network is created once per boot. Resetting it to collect network creation logs.
+   echo Deleting HNS network
+   powershell.exe -NoProfile "Get-HnsNetwork | Where-Object {$_.Name -eq 'WSL'} | Remove-HnsNetwork"
+
+   :: Stop WSL.
+   net.exe stop WslService || net.exe stop LxssManager
+)
+
+wsl.exe tr -d "\r" ^| bash < ../networking.sh > wsl_network_configuration_before.log
+powershell Get-NetRoute > get_netroute.log
+
+powershell invoke-expression 'cmd /c start powershell -Command { ..\collect-wsl-logs.ps1 }'
+powershell invoke-expression 'cmd /c start powershell -Command { "wsl.exe -u root sudo tcpdump -n -i any > tcpdump.log" }'
+
+wpr -start ..\wsl_networking.wprp -filemode -instanceName wpr_networking

```

Comment:
Instead of starting another log collection, can you pass this file to collect-wsl-logs.ps1 ?
 `collect-wsl-logs.ps1 -LogProfile wsl_networking.wprp` 

If needed we can copy all the providers in the default log file to `wsl_networking.wprp`. 

I think this would make a things a bit simpler, and collect-wsl-logs.ps1 has the benefit of handling the case where the log collection is already running

Reply:
"Instead of starting another log collection, can you pass this file to collect-wsl-logs.ps1" makes sense, will do.

"If needed we can copy all the providers in the default log file to wsl_networking.wprp" the latest iteration of the PR already does this

Reply:
updated wsl_networking.wprp and used wpr with wsl_networking.wprp (instead of calling collect-wsl-logs.ps1 from the new script)